src/write_gcm.c

```diff
@@ -3823,7 +3823,7 @@ static int wg_transmit_unique_segment(const wg_context_t *ctx,
 
     // By the way, a successful response is an empty JSON record (i.e. "{}").
     // An unsuccessful response is a detailed error message from the API.
-    char response[2048];
+    char response[8192];
```

Comment:
We can send ~200 points in each request. Let's assume the error message is 100 characters for each — that's already 20k. Shall we make the buffer 64k, just to be on the safe side?

Reply:
If the metric names are long it's actually a lot longer than 100 characters. I'd do 128K.

Reply:
Ling and I saw a pretty regular looking error message yesterday, of which only 1.5 fit in a response, so to be safe I'd go 200 * 2048. Can we get curl to do dynamic memory allocation?

It might be just easier to fix the bug...

Reply:
@dhrupadb thoughts on dynamic memory allocation? You're more well versed with C than I am.

Reply:
The curl docs for the CURLOPT_WRITEDATA flag point to this example of dynamic memory handling: https://curl.haxx.se/libcurl/c/getinmemory.html

Ref: https://curl.haxx.se/libcurl/c/CURLOPT_WRITEDATA.html

Reply:
An initial investigation seems to indicate this isn't possible. Managing the buffer's size ourselves is our best bet. Our options really are 1. Allocate a really large buffer and play it safe. Downside is that this isn't fool proof. 2.  The `wg_curl_write_callback` is changed to grow the buffer. However, right now the buffer is a stack allocated buffer. We'd have to make this a heap allocated buffer instead. See https://curl.haxx.se/libcurl/c/getinmemory.html on how to do this.

Reply:
Allocating 400k on the stack is audacious. It's not clear it'll always succeed. I would suggest bumping this down to 64k, which ought to cover most cases in the short term.
If heap-allocated buffers are available in the older versions of `libcurl` that we use, we can follow-up with a PR that uses that instead.

Reply:
I don't think heap allocation is a libcurl issue. We will need to send the memory buffer as a part of the `wg_write_ctx` struct no matter what. At the moment we choose to send memory on the stack. If we decide that it makes sense to not restrict the response and have the buffer size grow as needed, then we will have to dynamically allocate a conservative buffer initially, following which the `write_callback` will allocate a larger buffer if needed. That said, I'm fine with doing it in another PR.

Reply:
Ok, write callbacks (via `CURLOPT_WRITEFUNCTION`) were supported at least as far back as curl 7.18.0. The oldest OS we support today, CentOS 6, came with curl 7.19.7. All the other distros have newer versions of curl. So we should be able to implement that recipe.